tests/wicked/2nics/sut/t03_setup_second_card.pm

```diff
@@ -12,29 +12,32 @@
 #             Jose Lausuch <jalausuch@suse.com>
 #             Clemens Famulla-Conrad <cfamullaconrad@suse.de>
 
-use base 'wickedbase';
-use strict;
-use warnings;
+use Mojo::Base 'wickedbase';
 use testapi;
-use utils 'file_content_replace';
+use lockapi;
+use utils qw(file_content_replace systemctl);
 use network_utils 'iface';
 
 
 sub run {
     my ($self, $ctx) = @_;
-    my $config  = '/etc/sysconfig/network/ifcfg-' . $ctx->iface();
-    my $config2 = '/etc/sysconfig/network/ifcfg-' . $ctx->iface2();
+    mutex_wait('t03_dhcpd_setup_complete');
```

Comment:
would prefer to have it exactly before line where test should stop , this will be more clear to understand and debug thing later . I mean I sure that we can do everything till wicked restart without waiting for ref, right ?

Reply:
Ok, I will move that line before `ifup eth0`

Reply:
A other point I would like to discuss. There is the possibility that something goes wrong on REF before calling `mutex_create('t03_dhcpd_setup_complete');`. Then this test will run for 2h until some global timer will expire.
What do you think in adding a "wait_time" for `mutex_lock()`. Of course this is nothing for this PR, only in general.

Reply:
as far as I know this is handled by openQA backend . in this case SUT will get parallel failed status and end up immediately

Reply:
I'm going to create a poo so we do not forget about that.